django_rq/queues.py

```diff
@@ -181,21 +185,36 @@ def get_queues(*queue_names, **kwargs):
     All instances must use the same Redis connection.
     """
     from .settings import QUEUES
-    autocommit = kwargs.get('autocommit', None)
-    queue_class = kwargs.get('queue_class', DjangoRQ)
-    if len(queue_names) == 0:
+
+    if len(queue_names) <= 1:
         # Return "default" queue if no queue name is specified
-        return [get_queue(autocommit=autocommit)]
-    if len(queue_names) > 1:
-        queue_params = QUEUES[queue_names[0]]
-        connection_params = filter_connection_params(queue_params)
-        for name in queue_names:
-            if connection_params != filter_connection_params(QUEUES[name]):
-                raise ValueError(
-                    'Queues must have the same redis connection.'
-                    '"{0}" and "{1}" have '
-                    'different connections'.format(name, queue_names[0]))
-    return [get_queue(name, autocommit=autocommit, queue_class=queue_class) for name in queue_names]
+        # or one queue with specified name
+        return [get_queue(*queue_names, **kwargs)]
+
+    # will return more than one queue
+    # import job class only once for all queues
+    kwargs['job_class'] = get_job_class(kwargs.pop('job_class', None))
+
+    queue_params = QUEUES[queue_names[0]]
+    connection_params = filter_connection_params(queue_params)
+    queues = [get_queue(queue_names[0], **kwargs)]
+
+    # do consistency checks while building return list
+    for name in queue_names[1:]:
+        queue = get_queue(name, **kwargs)
+        if queue.__class__ is not queues[0].__class__:
```

Comment:
Please use `isinstance`

Reply:
As I said, `isinstance` is exactly what we **don't** want here. `get_queues` is mostly used for worker initialization and worker must get one queue class. If we want to run worker for two queues where one is of `DjangoRQ` class and another is some `CustomQueue` extending `DjangoRQ` we cannot guarantee that `CustomQueue` has the same implementation for functionality used by worker class.

Regarding `__class__`, I can change that to `type(queue)`. Would that be better?

Reply:
@skirsdeda you're right, I think :).

Yes, please change this to `type(queue)`